miniapp/miniapp.cpp

```diff
@@ -137,15 +118,35 @@ int main(int argc, char** argv) {
             m.attach_sampler(probe.id, make_trace_sampler(traces.back().get(), sample_dt));
         }
 
+#ifdef WITH_PROFILING
         // dummy run of the model for one step to ensure that profiling is consistent
         m.run(options.dt, options.dt);
-
-        // reset the model
+        // reset and add the source spikes once again
         m.reset();
-        // reset the source spikes
         for (auto source : local_sources) {
             m.add_artificial_spike({source, 0});
         }
+#endif
+
+        // file output depends on the input arguments set the output file after
+        // the dummy run to ensure that the initial "seed spikes" are recorded once
```

Comment:
This is not a dummy run to seed the spikes. It is a dummy run to initialize the profiler (makes a difference).

The reason that the io code was moved here was that if it is set before the dummy run, the spikes from the dummy run will be recorded and saved to disk... and then will be written again for the real run. So we hook up the io after the dummy run.